soc/nxp/s32/s32ze/soc.h

```diff
@@ -11,7 +11,7 @@
 #define __GIC_PRESENT 0
 
 #if defined(CONFIG_SOC_S32Z270)
-#include <S32Z2.h>
+#include "S32Z2_MC_CGM.h"
```

Comment:
>add the essential header files to build with mcux

please elaborate, why?

Reply:
+ In _"S32Z2.h"_ included _"S32Z2_SPI.h"_ that has the struct **SPI_Type** and the macros as same as which I has defined in  _modules\hal\nxp\s32\mcux\devices\S32Z270\S32Z270_device.h_ and **mcux dspi** is using, leads to raise error re-defined. If we want to build DSPI with mcux, we cannot include "S32Z2_SPI.h"

+ If we include <S32Z2.h>, we could add unnecessary files to build

That why I want to include the essential files

I propose 2 solutions:
+ do like which implemented
+ jus remove or guard _#include "S32Z2_SPI.h"_ in _modules\hal\nxp\s32\drivers\s32ze\BaseNXP\header\S32Z2.h_ and keep them all